Code/Source/Frame/ROS2FrameComponent.cpp

```diff
@@ -158,4 +173,13 @@ namespace ROS2
     {
         required.push_back(AZ_CRC("TransformService"));
     }
+
+    ROS2FrameComponent::ROS2FrameComponent()
+    {
+    }
+
+    ROS2FrameComponent::ROS2FrameComponent(const AZStd::string& frameId)
+        : m_frameName(frameId)
+    {
+    }
```

Comment:
This should accept the frameId parameter  by value and move it into the m_frameName to prevent a dynamic allocation of an `AZStd::string` when a string literal is created, followed by a copy operation to copy the `AZStd::string` to member variable.

```suggestion
    ROS2FrameComponent::ROS2FrameComponent(AZStd::string frameId)
        : m_frameName(AZStd::move(frameId))
    {
    }
```